plugins/modules/ios_vrf_interfaces.py

```diff
@@ -0,0 +1,469 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2024 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""
+The module file for ios_vrf_interfaces
+"""
+
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+module: ios_vrf_interfaces
+extends_documentation_fragment:
+  - cisco.ios.ios
+short_description: Manages VRF configuration on interfaces.
+description:
+  - Manages Virtual Routing and Forwarding (VRF) configuration on interfaces of Cisco IOS devices.
+version_added: "1.0.0"
+author: "AAYUSH ANAND (@AAYUSH2091)"
+notes:
+  - Tested against Cisco IOS XE Version 17.13.01a
+  - VRF must exist before assigning to an interface
+  - When removing VRF from interface, associated IP addresses will be removed
+options:
+  config:
+    description: A list of interface VRF configurations.
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+          - Full name of the interface to be configured.
+          - Example - GigabitEthernet1, Loopback24
+        type: str
+        required: true
+      vrf_name:
+        description:
+          - Name of the VRF to be configured on the interface.
+          - When configured, applies 'vrf forwarding <vrf_name>' under the interface.
+        type: str
+  running_config:
+    description:
+      - This option is used only with state I(parsed).
+      - The value of this option should be the output received from the IOS device by executing
+        the command B(show running-config | section ^interface).
+      - The state I(parsed) reads the configuration from C(running_config) option and transforms
+        it into Ansible structured data as per the resource module's argspec and the value
+        is then returned in the I(parsed) key within the result.
+    type: str
+  state:
+    description:
+      - The state the configuration should be left in.
+    type: str
+    choices:
+      - merged
+      - replaced
+      - overridden
+      - deleted
+      - gathered
+      - rendered
+      - parsed
+    default: merged
+"""
+
+EXAMPLES = """
```

Comment:
I don't see replaced  and deleted examples

Reply:
added in examples now